src/runtime_src/core/common/query_requests.h

```diff
@@ -1324,9 +1339,16 @@ struct flash_type : request
 {
   using result_type = std::string;
   static const key_type key = key_type::flash_type;
+  static const char* name() { return "flash_type"; }
 
   virtual boost::any
   get(const device*) const = 0;
+  
+  static std::string
+  to_string(result_type value)
+  {
+    return value;
+  }
```

Comment:
ditto.  Understood that it may be more flexible to use to_string on the query return value; it will allow us to change the return type if necessary.  So, maybe it's a  good idea after all.

Reply:
This function is used in device.h to populate the ptree:
```
template <typename ValueType>
static void
put(const ValueType& value, boost::property_tree::ptree& pt)
{
  pt.put(QueryRequestType::name(), QueryRequestType::to_string(value));
}
```